### PR TITLE
Utils.projectDelete(...), testing and verification

### PR DESCRIPTION
**Pull request by brunobowden**
Please review PR #312 first

- proj.delete(…) => Utils.projectDelete(…)
- Alows testing and verification of these calls
- Update the plugin with the switch
- Consolidate projectCopy, projectDelete and projectExec calls and
testing at the end of each class. Any moved method were left unchanged
and don’t need detailed review.

### CODE REVIEW

**Comment by brunobowden** [comment]
@advayDev1, @confile - This add proper testing around project.delete() calls as well. It needs a TypeCheckingMode.SKIP method, same as projectCopy and projectExec.

File: src/test/groovy/com/github/j2objccontrib/j2objcgradle/tasks/MockProjectExec.groovy [hunk 1]
```diff
@@ -62,10 +62,12 @@ class MockProjectExec {
 
         // This intercepts all methods, stubbing out exec and passing through all other invokes
         project.metaClass.invokeMethod = { String name, Object[] args ->
-            if (name == 'copy') {
+            // Call the proxy object so that it can track verifications
+            if (name == 'delete') {
+                return projectProxyInstance().delete(args.first())
```
Comment (advayDev1) [inline src/test/groovy/com/github/j2objccontrib/j2objcgradle/tasks/MockProjectExec.groovy]: cast to Closure?

Comment (advayDev1) [inline src/test/groovy/com/github/j2objccontrib/j2objcgradle/tasks/MockProjectExec.groovy]: uh no, cast to Object[] can't be sure

**Comment by advayDev1** [comment]
lgtm

File: src/test/groovy/com/github/j2objccontrib/j2objcgradle/tasks/MockProjectExec.groovy (hunk 1 above)
Comment (brunobowden) [inline src/test/groovy/com/github/j2objccontrib/j2objcgradle/tasks/MockProjectExec.groovy]: I didn't understand this comment?

Comment (advayDev1) [inline src/test/groovy/com/github/j2objccontrib/j2objcgradle/tasks/MockProjectExec.groovy]: sorry, this is a confusion over how varargs works with invokeMethod.
the parameters are delete(Object... args).
does args.first() get the array Object[], or just the first param as an Object?  do you instead need to just pass the whole args?

Comment (advayDev1) [inline src/test/groovy/com/github/j2objccontrib/j2objcgradle/tasks/MockProjectExec.groovy]: @brunobowden - please do answer my question above

Comment (brunobowden) [inline src/test/groovy/com/github/j2objccontrib/j2objcgradle/tasks/MockProjectExec.groovy]: For `proj.delete('DEL-1', 'DEL-2')` it gets ['DEL-1', 'DEL-2']. The unit test in UtilsTest very specifically covers varargs:

    @Test
    // Tests intercepting and verifying call to project.delete(path1, path2)
    void testProjectDelete_MockProjectExec() {
        MockProjectExec mockProjectExec = new MockProjectExec(proj, '/J2OBJC_HOME')

        mockProjectExec.demandDeleteAndReturn('/PATH1', '/PATH2')

        Utils.projectDelete(proj, '/PATH1', '/PATH2')

        mockProjectExec.verify()
    }

Comment (brunobowden) [inline src/test/groovy/com/github/j2objccontrib/j2objcgradle/tasks/MockProjectExec.groovy]: To be more specific, a call to `proj.delete('DEL-1', 'DEL-2')` means that in invokeMethod `args == [['DEL-1', 'DEL-2']]`, then it passes `['DEL-1', 'DEL-2']` to `projectProxyInstance().delete(args.first())`.

Comment (advayDev1) [inline src/test/groovy/com/github/j2objccontrib/j2objcgradle/tasks/MockProjectExec.groovy]: thanks!

**Pull request merged**